src/utils/tokens.ts

```diff
@@ -0,0 +1,138 @@
+import { Cluster, Connection, PublicKey } from "@solana/web3.js";
+import {
+  ENV,
+  Strategy,
+  TokenInfo,
+  TokenListContainer,
+  TokenListProvider,
+} from "@solana/spl-token-registry";
+import { deserializeMint } from "@saberhq/token-utils";
+import {
+  STEP_SWAP_PROGRAM_ID,
+  TokenSwap as StepTokenSwap,
+  TokenSwapLayout as StepSwapLayout,
+} from "@stepfinance/step-swap";
+import axios from "axios";
+import { getMultipleAccounts } from "./web3";
+
+// TODO: Update @stepfinance/step-swap to export this
+export const STEP_SWAP_OWNER = new PublicKey(
+  "GkT2mRSujbydLUmA178ykHe7hZtaUpkmX2sfwS8suWb3"
+);
+
+// shorten the checksummed version of the input address to have 4 characters at start and end
+export function shortenAddress(address: string, chars = 4): string {
+  return `${address.slice(0, chars)}...${address.slice(-chars)}`;
+}
+
+export type TokenMap = {
+  [address: string]: TokenInfo;
+};
+
+export const getTokenMap = async (
+  connection: Connection,
+  cluster: Cluster
+): Promise<TokenMap> => {
+  const baseTokenListProvider = new TokenListProvider();
+  const baseTokenList = await baseTokenListProvider.resolve(Strategy.GitHub);
+  const baseTokenInfos = baseTokenList
+    .filterByClusterSlug(cluster)
+    .excludeByTag("lp-token")
+    .excludeByTag("tokenized-stock")
+    .getList();
+
+  const tokenMap: TokenMap = baseTokenInfos.reduce<TokenMap>(
+    (map, tokenInfo) => ({ ...map, [tokenInfo.address]: tokenInfo }),
+    {}
+  );
+
+  const { data: rawStepTokenOverridesList } = await axios.get<TokenInfo[]>(
```

Comment:
Is it possible this can throw if it can't make the connection? Should we try/catch and handle a little better here?